src/OpenTelemetry/Sdk.cs

```diff
@@ -32,7 +34,24 @@ namespace OpenTelemetry
     /// </summary>
     public static class Sdk
     {
-        private static TimeSpan defaultPushInterval = TimeSpan.FromSeconds(60);
+        private static readonly TimeSpan DefaultPushInterval = TimeSpan.FromSeconds(60);
+
+        private static readonly RuntimeContextSlot<bool> SuppressInstrumentationRuntimeContextSlot = RuntimeContext.RegisterSlot<bool>("otel.suppress_instrumentation");
+
+        public static bool SuppressInstrumentation
```

Comment:
* We should have XML comments here. Important spot :) Not sure exactly how to word it though?
```csharp
		/// <summary>
		/// Gets or sets a value indicating whether or not <see cref="Activity"/>Activity</see> object collection in the current context should be suppressed (dsiabled). Default value: False.
		/// </summary>
		/// <remarks>
		/// Set <see cref="SuppressInstrumentation"/> to <see langword="true"/> when you want to turn off automatic <see cref="Activity"/> collection.
		/// This is typically done to prevent infinte loops created by collection of internal operations, such as exporting spans over HTTP.
		/// <code>
		///    public override async Task&lt;ExportResult&gt; ExportAsync(IEnumerable&lt;Activity&gt; batchActivity, CancellationToken cancellationToken)
		///    {
		///       Sdk.SuppressInstrumentation = true;
		///       try
		///       {
		///          await this.SendBatchActivityAsync(batchActivity, cancellationToken).ConfigureAwait(false);
		///          return ExportResult.Success;
		///       }
		///       finally
		///       {
		///          Sdk.SuppressInstrumentation = false;
		///       }
		///    }
		/// </code>
		/// </remarks>
```
* Should context be in the prop name? `SuppressInstrumentationOnCurrentContext`? Trying to hint to users what it is doing.

Reply:
> * We should have XML comments here. Important spot :) Not sure exactly how to word it though?

Good catch, will add it.

> * Should context be in the prop name? `SuppressInstrumentationOnCurrentContext`? Trying to hint to users what it is doing.

Not sure, the proposed name seems too long 😅
Perhaps the XML comment should address this?

Reply:
What if it wasn't exposed as a bool? Something more like log scope: `public IDisposable SuppressInstrumentation()`
Which is to say, return an object that sets the bool to true on ctor, sets it to fault on dispose?

Reply:
I believe this is what @reyang was proposing in the PR description. I think I do like the disposable idea. I'm currently playing with this implementation and I thing this would make things read nicer.

Reply:
Yeah, @alanwest is helping on that part in a separate PR ❤️